tests/test_train.py

```diff
@@ -358,6 +363,258 @@ def test_train_nlu_autoconfig(
+        default_stack_config: Text,
+        default_e2e_stories_file: Text,
+        default_nlu_data: Text,
+    ):
+        nlu_yaml = rasa.shared.utils.io.read_yaml_file(default_nlu_data)
+        nlu_yaml["nlu"][0]["examples"] += "- surprise!\n"
+
+        new_nlu_file = tempfile.NamedTemporaryFile(mode="w", suffix=".yml").name
+
+        rasa.shared.utils.io.write_yaml(nlu_yaml, new_nlu_file)
+
+        mocked_nlu_training = mock_nlu_training(monkeypatch)
+        mocked_core_training = mock_core_training(monkeypatch)
+
+        new_model_path = train(
+            default_domain_path,
+            default_stack_config,
+            [default_e2e_stories_file, new_nlu_file],
+            output=new_model_path_in_same_dir(trained_e2e_model),
+        ).model
+        os.remove(new_model_path)
+
+        mocked_core_training.assert_called_once()
+        mocked_nlu_training.assert_called_once()
+
+    def test_new_nlu_data_does_not_retrain_core_if_there_are_no_e2e_stories(
+        self,
+        monkeypatch: MonkeyPatch,
+        trained_simple_rasa_model: Text,
+        default_domain_path: Text,
+        default_stack_config: Text,
+        simple_stories_file: Text,
+        default_nlu_data: Text,
+    ):
+        nlu_yaml = rasa.shared.utils.io.read_yaml_file(default_nlu_data)
+        nlu_yaml["nlu"][0]["examples"] += "- surprise!\n"
+
+        new_nlu_file = tempfile.NamedTemporaryFile(mode="w", suffix=".yml").name
+
+        rasa.shared.utils.io.write_yaml(nlu_yaml, new_nlu_file)
+
+        mocked_nlu_training = mock_nlu_training(monkeypatch)
+        mocked_core_training = mock_core_training(monkeypatch)
+
+        new_model_path = train(
+            default_domain_path,
+            default_stack_config,
+            [simple_stories_file, new_nlu_file],
+            output=new_model_path_in_same_dir(trained_simple_rasa_model),
+        ).model
+        os.remove(new_model_path)
+
+        mocked_core_training.assert_not_called()
+        mocked_nlu_training.assert_called_once()
+
+    def test_training_core_with_e2e_fails_gracefully(
+        self,
+        capsys: CaptureFixture,
+        monkeypatch: MonkeyPatch,
+        tmp_path: Path,
+        default_domain_path: Text,
+        default_stack_config: Text,
+        default_e2e_stories_file: Text,
+    ):
+
+        mocked_nlu_training = mock_nlu_training(monkeypatch)
+        mocked_core_training = mock_core_training(monkeypatch)
+
+        output = self.make_tmp_model_dir(tmp_path)
+        train_core(
+            default_domain_path,
+            default_stack_config,
+            default_e2e_stories_file,
+            output=output,
+        )
+
+        mocked_core_training.assert_not_called()
+        mocked_nlu_training.assert_not_called()
+
+        captured = capsys.readouterr()
+        assert (
+            "Stories file contains e2e stories. "
+            + "Please train using `rasa train` so that the NLU model is also trained."
```

Comment:
```suggestion
            "Please train using `rasa train` so that the NLU model is also trained."
```
```suggestion
            + "Please train using `rasa train` so that the NLU model is also trained."
```